source/agora/consensus/protocol/Nominator.d

```diff
@@ -179,6 +179,14 @@ extern(D):
         if (data.tx_set.length == 0)
             return;  // not ready yet
 
+        // check whether the consensus data is valid before nominating it.
+        if (auto msg = this.ledger.validateConsensusData(data))
```

Comment:
Although I question why this check is in here rather than inside of `prepareNominatingSet`. It seems like leaky abstractions?